Source/UI/NewUI/PaymentForm/View/CardNumber/CardNumberView.swift

```diff
@@ -73,18 +77,27 @@ extension CardNumberView: TextFieldViewDelegate {
   func textFieldShouldReturn() -> Bool { true }
 
   func textFieldShouldEndEditing(textField: UITextField, replacementString: String) -> Bool {
+    textField.text = replacementString
+    style?.textfield.text = replacementString
     switch viewModel.validate(cardNumber: replacementString) {
     case .some(let schemeIcon):
       self.schemeIcon = schemeIcon
       updateError(show: false)
     case nil:
       updateError(show: true)
     }
-
     return true
   }
 
   func textField(_ textField: UITextField, shouldChangeCharactersIn range: NSRange, replacementString string: String) -> Bool {
+
+    // some characters are deleted
+    // notify payment view controller to disable the pay button
+    if string.count == 0 && range.length > 0 {
+      delegate?.textFieldDidStartDeleteText()
+      return true
+    }
+
```

Comment:
shouldn't this just result in a standard update to the input which then triggers a validation that will obviously disable the button? Why do we need to say "deleted"? 

Fyi, this will also break if you happen to have some card scheme that has consecutive valid number of inputs, or for VISA as its been implemented to handle 19 characters. 17 is not valid, but deleting another to 16 may have valid number. 

This logic, assuming a char delete will result in invalid input, is inherently flawed.

_Also I will only leave 1 comment here, but same applies to every input field in this PR._

Reply:
what about validating firstly then if not valid then fire the delegate ?

Reply:
@alex-ioja-yang-cko what about this suggestion ?

```suggestion
    // some characters are deleted
    // notify payment view controller to disable the pay button
    if string.count == 0 && range.length > 0 {
       if let text = textField.text, viewModel.validate(cardNumber: text) == nil {
          delegate?.textFieldDidStartDeleteText()
       }
       return true
   }
```